### PR TITLE
Add more profiling and benchmarking documentation

### PR DESCRIPTION
**Pull request by glebm**
On Linux, you can now run:

```
tools/build_and_run_benchmark.py --gperf devilutionx -- --diablo --spawn --lang en --demo 0 --timedemo
```

and get this:
image
Screenshot from 2024-08-09 16-05-30
Screenshot from 2024-08-09 16-05-53

### CODE REVIEW

**Comment by qndel** [comment]
https://godbolt.org/z/jfecMrx43
This doesn't look good :P wrong values + Program terminated with signal: SIGFPE

File: Source/engine/render/clx_render.cpp [hunk 1]
```diff
@@ -77,16 +77,18 @@ struct SkipSize {
...
-	}
-	// If `remainingWidth` is non-negative, then it is 0, meaning we drew a whole line.
-	return { 1, 0 };
+	// If `remainingWidth` is negative, `-remainingWidth` is the overrun.
+	// Otherwise, `remainingWidth` is always 0.
+
+	// Remaining width of 0 (= no overrun) is a common case.
+	// The calculation below would result in the same result.
+	// However, checking for 0 and skipping it entirely turns out to be faster.
+	if (remainingWidth == 0) return { 1, 0 };
```
Comment (qndel) [inline Source/engine/render/clx_render.cpp:86]:
```suggestion
	if (remainingWidth >= 0) return { 1, 0 };
```

Comment (glebm) [reply Source/engine/render/clx_render.cpp:86]: That uses a different `GetSkipSize` though.

Comment (qndel) [reply Source/engine/render/clx_render.cpp]: > That uses a different `GetSkipSize` though.

oh well didn't check that far 😃

**Review by AJenbo — APPROVED** [review]

**Pull request merged**